extensions/amp-iframe/0.1/amp-iframe.js

```diff
@@ -105,7 +105,14 @@ export class AmpIframe extends AMP.BaseElement {
         !((' ' + sandbox + ' ').match(/\s+allow-same-origin\s+/i)),
         'allow-same-origin is not allowed with the srcdoc attribute %s.',
         this.element);
-    return 'data:text/html;charset=utf-8;base64,' + btoa(srcdoc);
+
+    // srcdoc is a DOMString and requires special handling for base64 encoding.
+    // See https://developer.mozilla.org/en-US/docs/Web/API/WindowBase64/Base64_encoding_and_decoding
+    const encodedSrcdoc = btoa(encodeURIComponent(srcdoc)
```

Comment:
@muxin @jridgewell Here is what I am thinking to do:

1- Change the iframe to use `utf8Encode` from `bytes.js` and then pass it to `base64EncodeFromBytes` from `base64.js`. This should be the fastest way to do this specially for browsers that have native TextEncoder support.

2- Remove `base64Encode` and `base64Decode` from `base64.js` They are not used anywhere and are confusing (same way `btoa` is confusing) in terms of what input string are valid. 

thoughts?